src/io.c

```diff
@@ -137,6 +137,48 @@ static int pids[MAXCHLDS];         /* and this number! */
 static int fistats = 0;            /* First used entry */
 static int lastats = 0;            /* First unused entry */
 static int statsfull = 0;          /* Flag, whether stats FIFO full */
+
+static int findSignaledPid(int pidc)
+{
+    int pos = fistats;
+    do {
+        if (pids[pos] == pidc) break;
```

Comment:
Perhaps replace this odd infinite-loop-with-exit-check-at-start by a more idiomatic `while` loop:
```
while (pids[pos] != pidc) {
...
}
```